eq-author-api/db/index.js

```diff
@@ -1,4 +1,45 @@
-var environment = process.env.NODE_ENV || "development";
-var config = require("../config/knexfile.js")[environment];
+const config = require("../config/knexfile.js");
 
-module.exports = require("knex")(config);
\ No newline at end of file
+const getConfig = async function(secretId) {
+  if (secretId && secretId !== "") {
+    const AWS = require("aws-sdk");
+    const client = new AWS.SecretsManager({ region: "eu-west-1" });
+
+    let secretPromise = client.getSecretValue({ SecretId: secretId }).promise();
+
+    try {
+      let secret = await secretPromise;
+      let dbCredentials = JSON.parse(secret.SecretString);
+      config.connection = {
+        host: dbCredentials.host,
+        port: dbCredentials.port,
+        user: dbCredentials.username,
+        password: dbCredentials.password,
+        database: dbCredentials.dbname
+      };
+      return config;
```

Comment:
Instead of mutating the `config` object (which is imported). I think it would be cleaner to define a locally scoped variable to hold the configuration and have a single return at the end of the function.

e.g. 

```js
const getConfig = () => {
 const result = {
   ...config
 }

 if (hasAWSCreds) {
   result.connection = ...
 }


 return result;
}

```

Reply:
👍